pkg/runner/collector_selector.go

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+package runner
+
+import (
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/redhat-partner-solutions/vse-sync-testsuite/pkg/collectors"
+)
+
+var (
+	optionalCollectorNames []string
+	requiredCollectorNames []string
+	All                    string = "all"
+)
+
+func init() {
+	optionalCollectorNames = make([]string, 0)
+	optionalCollectorNames = append(optionalCollectorNames,
+		collectors.DPLLCollectorName,
+		collectors.GPSCollectorName,
+	)
+
+	requiredCollectorNames = make([]string, 0)
+	requiredCollectorNames = append(requiredCollectorNames,
+		collectors.DevInfoCollectorName,
+	)
+}
+
+func isIn(name string, arr []string) bool {
+	for _, arrVal := range arr {
+		if name == arrVal {
+			return true
+		}
+	}
+	return false
+}
+
+func removeDuplicates(arr []string) []string {
+	res := make([]string, 0)
+	for _, name := range arr {
+		if !isIn(name, res) {
+			res = append(res, name)
+		}
+	}
+	return res
+}
+
+func GetCollectorsNames(selectedCollectors []string) []string {
```

Comment:
This one needs a bit of documentation as it's doing several things, and perhaps a more descriptive name e.g. `GetCollectorsToRun`.